### PR TITLE
Added an example of using the proc filesystem (add more linux kernel programs #2)

### PR DESCRIPTION
**Pull request by d-grossman**
There is some weird involved with the copy_to_user call. I'm getting reasonable values going into the call, and then an oops.

### CODE REVIEW

File: procfs_hw/procfs_hw.c [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	printk("proc called read %ld\n", count);
+	printk("proc called len %ld\n", len);
+	printk("about to copy %s\n", message);
+
+	/*
+	 * should be len instead of 0, but that causes an oops
+	 */
+
+	/* @todo figure out why copy_to_user so upset wich nonzero */
+	copy_to_user(buf, message, 0);
```
Comment (d-grossman) [inline procfs_hw/procfs_hw.c]: this should be 
copy_to_user(buf, message, len)
I'm not sure if it is my vm that is bad or a real bug.

**Comment by d-grossman** [comment]
example works,
```
user@ubuntu:~/x$ uname -a
Linux ubuntu 4.4.0-97-generic #120-Ubuntu SMP Tue Sep 19 17:28:18 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```
```
user@ubuntu:~/x$ make
make -C /lib/modules/4.4.0-97-generic/build M=/home/user/x modules
make[1]: Entering directory '/usr/src/linux-headers-4.4.0-97-generic'
  CC [M]  /home/user/x/procfs_hw.o
  Building modules, stage 2.
  MODPOST 1 modules
  CC      /home/user/x/procfs_hw.mod.o
  LD [M]  /home/user/x/procfs_hw.ko
make[1]: Leaving directory '/usr/src/linux-headers-4.4.0-97-generic'
user@ubuntu:~/x$ sudo insmod procfs_hw.ko
user@ubuntu:~/x$ cat /proc/procfs_hw 
Hello World

user@ubuntu:~/x$ sudo rmmod procfs_hw 
```

**Comment by d-grossman** [comment]
@doomers anything else to close this out?

**Comment by doomers** [comment]
nopes great work, actually i was not active from last 2-3 days on github.
Thanks for contributing

**Pull request merged**